python/cugraph/cugraph/dask/sampling/uniform_neighbor_sample.py

```diff
@@ -268,4 +279,7 @@ def uniform_neighbor_sample(
         ddf = input_graph.unrenumber(ddf, "sources", preserve_order=True)
         ddf = input_graph.unrenumber(ddf, "destinations", preserve_order=True)
 
+    if _multiple_clients:
+        lock.release()
```

Comment:
Could the lock be released earlier to improve parallelism slightly, like right before the unrenumbering above (esp. since I think unrenumbering could be expensive)?

Reply:
@jnke2016 , Do we use PLC primitives to unrenumbe, I believe those are not multi-client proof so i dont think that will safe. 

If it is a pure dask operation we can do it safely.

Reply:
I believe unrenumbering does not use PLC and is pure cudf/dask_cudf code, but @jnke2016 should also comment.

Reply:
@rlratzel correct: `unrenumbering` is just pure cudf/dask_cudf

Reply:
Thanks pushed latests changes with this assumption.